Move version.h to torch/headeronly

mikaylagawarecki opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #164381

Differential Revision: [D83685392](https://our.internmc.facebook.com/intern/diff/D83685392)

mikaylagawarecki committed:
Move version.h to torch/headeronly

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"




[ghstack-poisoned]

mikaylagawarecki commented on caffe2/CMakeLists.txt:
```diff
@@ -244,7 +244,7 @@ configure_file("${TORCH_SRC_DIR}/_utils_internal.py"
 
 # Generate header with version info
 configure_file("${TORCH_SRC_DIR}/csrc/api/include/torch/version.h.in"
-  "${TORCH_SRC_DIR}/csrc/api/include/torch/version.h"
```
we do below

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"




[ghstack-poisoned]

mikaylagawarecki commented:
@mikaylagawarecki has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D83685392).

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"


Differential Revision: [D83685392](https://our.internmc.facebook.com/intern/diff/D83685392)

[ghstack-poisoned]

janeyx99 commented on .gitignore:
```diff
@@ -88,7 +88,7 @@ torch_compile_debug/
 # Listed manually because some files in this directory are not generated
 torch/testing/_internal/generated/annotated_fn_args.py
 torch/testing/_internal/data/*.pt
-torch/csrc/api/include/torch/version.h
```
we should add this back then?

janeyx99 replied:
nvm!!

janeyx99 reviewed (approved):
LGTM conditional on CI passing

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"


Differential Revision: [D83685392](https://our.internmc.facebook.com/intern/diff/D83685392)

[ghstack-poisoned]

mikaylagawarecki commented:
@mikaylagawarecki has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D83685392).

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"


Differential Revision: [D83685392](https://our.internmc.facebook.com/intern/diff/D83685392)

[ghstack-poisoned]

mikaylagawarecki commented:
@mikaylagawarecki has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D83685392).

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"


Differential Revision: [D83685392](https://our.internmc.facebook.com/intern/diff/D83685392)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"


Differential Revision: [D83685392](https://our.internmc.facebook.com/intern/diff/D83685392)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"


Differential Revision: [D83685392](https://our.internmc.facebook.com/intern/diff/D83685392)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"


Differential Revision: [D83685392](https://our.internmc.facebook.com/intern/diff/D83685392)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"


Differential Revision: [D83685392](https://our.internmc.facebook.com/intern/diff/D83685392)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"


Differential Revision: [D83685392](https://our.internmc.facebook.com/intern/diff/D83685392)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"


Differential Revision: [D83685392](https://our.internmc.facebook.com/intern/diff/D83685392)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"


Differential Revision: [D83685392](https://our.internmc.facebook.com/intern/diff/D83685392)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"


Differential Revision: [D83685392](https://our.internmc.facebook.com/intern/diff/D83685392)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"


Differential Revision: [D83685392](https://our.internmc.facebook.com/intern/diff/D83685392)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"


Differential Revision: [D83685392](https://our.internmc.facebook.com/intern/diff/D83685392)

[ghstack-poisoned]

mikaylagawarecki committed:
Update on "Move version.h to torch/headeronly"


Differential Revision: [D83685392](https://our.internmc.facebook.com/intern/diff/D83685392)

[ghstack-poisoned]
